File: LeetCode/Move_Zeros.cpp
/*
Problem Link:
https://leetcode.com/problems/move-zeroes/description/
*/


class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int len = nums.size();
        int writeIter = 0;
        int readIter = 0;
        for(readIter = 0; readIter<len;readIter++){
            if(nums[readIter] != 0){
                nums[writeIter] = nums[readIter];
                writeIter++;
            }
            else{
                continue;
            }
        }
        for(;writeIter<len;writeIter++){
            nums[writeIter] = 0;
        }
    }
};

File: LeetCode/move_zeros.java
class Solution {
    public void moveZeroes(int[] nums) {
        int writeIterator = 0;
        for(int readIterator = 0;readIterator < nums.length; readIterator++){
            if(nums[readIterator] != 0){
                nums[writeIterator] = nums[readIterator];
                writeIterator++;
            }
        }
        for(; writeIterator < nums.length; writeIterator++){
            nums[writeIterator] = 0;
        }
    }
}